### PR TITLE
BACK-1698: Integrate wUSDM

### PR DESCRIPTION
**Pull request by JackieJoo**

### CODE REVIEW

File: src/dex/wusdm/wusdm.ts [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    return {
+      targetExchange: exchange,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // This is optional function in case if your implementation has acquired any resources
+  // you need to release for graceful shutdown. For example, it may be any interval timer
+  releaseResources(): AsyncOrSync<void> {
```
Comment (aburkut) [inline src/dex/wusdm/wusdm.ts:311]: @JackieJoo let's remove this empty method

File: src/dex/wusdm/wusdm.ts [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    );
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  isAppropriatePair(srcToken: Token, destToken: Token): boolean {
+    return (
+      (srcToken.address.toLowerCase() === this.wUSDMAddress.toLowerCase() &&
```
Comment (aburkut) [inline src/dex/wusdm/wusdm.ts:78]: @JackieJoo Can we use isUSDM and isWUSDM methods here?

File: src/dex/wusdm/wusdm.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        : undefined,
+      skipApproval: isSell && this.isUSDM(destToken),
+    };
+  }
+
+  // Encode params required by the exchange adapter
+  // V5: Used for multiSwap, buy & megaSwap
+  // V6: Not used, can be left blank
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
```
Comment (aburkut) [inline src/dex/wusdm/wusdm.ts:281]: @JackieJoo I think we can left this method empty since we are not going to support this integration on v5 I assume.

**Review by aburkut — APPROVED** [review]

File: src/dex/wusdm/wusdm.ts (hunk 1 above)
Comment (JackieJoo) [reply src/dex/wusdm/wusdm.ts]: fixed

File: src/dex/wusdm/wusdm.ts (hunk 2 above)
Comment (JackieJoo) [reply src/dex/wusdm/wusdm.ts]: done

**Review by 0xNazarii — APPROVED** [review]

**Pull request merged**